go-controller/pkg/ovn/pods.go

```diff
@@ -140,23 +141,22 @@ func (oc *Controller) deleteLogicalPort(pod *kapi.Pod) {
 
 func (oc *Controller) addLogicalPort(pod *kapi.Pod) {
 
-	count := 30
 	logicalSwitch := pod.Spec.NodeName
-	for count > 0 {
-		if logicalSwitch != "" {
-			break
-		}
-		if count != 30 {
-			time.Sleep(1 * time.Second)
-		}
-		count--
+	// Exponential back off ~32 seconds + 7* t(api call)
+	var getPodBackoff = wait.Backoff{Duration: 1 * time.Second, Steps: 7, Factor: 1.5, Jitter: 0.1}
+	if err := wait.ExponentialBackoff(getPodBackoff, func() (bool, error) {
 		p, err := oc.Kube.GetPod(pod.Namespace, pod.Name)
 		if err != nil {
-			logrus.Errorf("Could not get pod %s/%s for obtaining the logical switch it belongs to", pod.Namespace, pod.Name)
-			continue
+			logrus.Warningf("Could not get pod %s/%s for obtaining the logical switch it belongs to", pod.Namespace, pod.Name)
+			return false, nil
 		}
 		logicalSwitch = p.Spec.NodeName
+		return true, nil
```

Comment:
I think the logic has changed now from before. If logicalSwitch is empty, we will return as "true" and the retry will fail.

Reply:
It may make sense to revert the change from pods.go, since it does not contribute with DDOS.